qiskit_quantuminspire/qi_jobs.py

```diff
@@ -222,9 +222,11 @@ async def _fetch_job_results(self) -> None:
                 circuit_data.results = None if not result_item else result_item[0]
 
     @cache
-    def result(self) -> Result:
+    def result(self, wait_for_results: Optional[bool] = True, timeout: float = 60.0) -> Result:
         """Return the results of the job."""
-        if not self.done():
+        if wait_for_results:
+            self.wait_for_final_state(timeout=timeout)
```

Comment:
The `self.wait_for_final_state` is a builtin qiskit function and it will raise a `JobTimeoutError` when timeout occurs, so it will not loop infinitely. The polling logic in e2e test is used here as well, with timeout being the total time it will look for results and it polls every 5 secs by default

```
  def wait_for_final_state(
      self, timeout: Optional[float] = None, wait: float = 5, callback: Optional[Callable] = None
  ) -> None:
      """Poll the job status until it progresses to a final state such as ``DONE`` or ``ERROR``.

      Args:
          timeout: Seconds to wait for the job. If ``None``, wait indefinitely.
          wait: Seconds between queries.
          callback: Callback function invoked after each query.
              The following positional arguments are provided to the callback function:

              * job_id: Job ID
              * job_status: Status of the job from the last query
              * job: This BaseJob instance

              Note: different subclass might provide different arguments to
              the callback function.

      Raises:
          JobTimeoutError: If the job does not reach a final state before the
              specified timeout.
      """
      if not self._async:
          return
      start_time = time.time()
      status = self.status()
      while status not in JOB_FINAL_STATES:
          elapsed_time = time.time() - start_time
          if timeout is not None and elapsed_time >= timeout:
              raise JobTimeoutError(f"Timeout while waiting for job {self.job_id()}.")
          if callback:
              callback(self.job_id(), status, self)
          time.sleep(wait)
          status = self.status()
      return
```

Reply:
I added a test to make sure we don't loop infinitely